### PR TITLE
Added allocation of SYCL tables for BF KNN in case of GPU run (and other minor changes)

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_result.h [hunk 1]
```diff
@@ -52,20 +53,48 @@ DAAL_EXPORT services::Status Result::allocate(const daal::algorithms::Input * in
...
+                data_management::HomogenNumericTable<algorithmFPType>::create(1, nRows, data_management::NumericTableIface::doAllocate, &s));
+        }
+        else
+        {
+            set(prediction, data_management::internal::SyclHomogenNumericTable<algorithmFPType>::create(
+                                1, nRows, data_management::NumericTableIface::doAllocate, &s));
+        }
     }
 
     if (s.ok() && (par->resultsToCompute & computeIndicesOfNeighbors))
```
Comment (PetrovKP) [inline cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_result.h:73]: For gpu two next case not available now. Maybe we need to left only cpu branch?

**Review by PetrovKP — APPROVED** [review]

**Review by KulikovNikita — APPROVED** [review]

**Review by michael-smirnov — APPROVED** [review]

File: cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_dense_default_batch_container.h [hunk 2]
```diff
@@ -53,7 +54,7 @@ BatchContainer<algorithmFpType, method, cpu>::~BatchContainer()
 template <typename algorithmFpType, Method method, CpuType cpu>
 services::Status BatchContainer<algorithmFpType, method, cpu>::compute()
 {
-    const daal::algorithms::Parameter * const par            = _par;
+    const Parameter * const par                              = static_cast<const Parameter *>(_par);
```
Comment (michael-smirnov) [inline cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_dense_default_batch_container.h:57]: Why you need this cast?

Comment (amgrigoriev) [reply cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_dense_default_batch_container.h:57]: _par is a pointer to base class. I need a pointer to derived class to access par->resultsToEvaluate

**Pull request merged**